commit 281e0cbe9819786778738d9bb61772f1a57f5fbc (refs/changes/32/1332/3)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2017-01-25T17:43:00+00:00 (2 years, 8 months ago)
    
    MANTA-3116 disable pwritev in nginx
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/auto/unix b/auto/unix
index 1316e54a..a668b5cb 100755
--- a/auto/unix
+++ b/auto/unix
@@ -643,7 +643,14 @@ ngx_feature_test="char buf[1]; struct iovec vec[1]; ssize_t n;
                   vec[0].iov_len = 1;
                   n = pwritev(1, vec, 1, 0);
                   if (n == -1) return 1"
-. auto/feature
+#
+# Unfortunately, there are bugs in the implementation of pwritev on
+# illumos when building a 32-bit version with LFS support. Effectively,
+# we'll spuriously return EINVAL when writing at offsets between 2 GiB
+# and 4 GiB. When we have rolled out a fix for OS-5904 everywhere, then
+# we can remove this change.
+#
+# . auto/feature
 
 
 ngx_feature="sys_nerr"
